engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -748,7 +750,25 @@ namespace dmGameObject
                     return luaL_error(L, "Trying to set property value for '%s' with an index < 0: %d", dmHashReverseSafe64(property_id), property_options.m_Index);
                 }
             }
+            lua_pop(L, 1);
 
+            lua_getfield(L, -1, "key");
+            property_options.m_Key = 0;
+            if (!lua_isnil(L, -1))
+            {
+                if (lua_isstring(L, -1))
+                {
+                    property_options.m_Key = dmHashString64(lua_tostring(L, -1));
+                }
+                else if (dmScript::IsHash(L, -1))
+                {
+                    property_options.m_Key = dmScript::CheckHash(L, -1);
+                }
+                else
+                {
+                    return luaL_error(L, "Invalid key argument in options table. It should be string or hash.");
+                }
+            }
```

Comment:
I think we should handle the fact that this code kan use index+key at the same time. I don't think we want that.

Reply:
so we should throw Lua error if there are key AND index in this table at the same time?

Reply:
done